FileSystem/FileMapperMocker.js

```diff
@@ -10,28 +25,490 @@ var USE_FILEMAPPER_MOCKER = true;
+ * @param {Array} sourceFilePaths List of local files to be moved
+ * @param {string} targetFolderPath Destination folder path
+ */
+function moveFile(sourceFilePaths, targetFolderPath) {
+  var targetFolderId;
+  // Get folder, if it doesn't exist then create it
+  try {
+    targetFolderId = this.getFolderId(targetFolderPath);
+  } catch (e) {
+    targetFolderId = this.createFolder(targetFolderPath);
+  }
+  var targetFolder = DriveApp.getFolderById(targetFolderId);
+  // Move all files
+  for (var i = 0; i < sourceFilePaths.length; i++) {
+    var fileId = this.getFileId(sourceFilePaths[i]);
+    var file = DriveApp.getFileById(fileId);
+    file.makeCopy(file.getName(), targetFolder);
+    file.setTrashed(true);
+  }
+}
+
+/**
+ * Move folder API
+ * API moves all folders to the target folder. Apps Script does
+ * not have a move API. Move is implemented by making a copy
+ * and deleting the original folder. Apps Script does not have a
+ * direct copy folder API as well, a copy is done recursively
+ * @param {Array} sourceFilePaths List of local folders to be moved
+ * @param {string} targetFolderPath Destination folder path
+ */
+function moveFolder(sourceFilePaths, targetFolderPath) {
+  var targetFolderId;
+  // Get folder, if it doesn't exist then create it
+  try {
+    targetFolderId = this.getFolderId(targetFolderPath);
+  } catch (e) {
+    targetFolderId = this.createFolder(targetFolderPath);
+  }
+  var targetFolder = DriveApp.getFolderById(targetFolderId);
+  // Move all folders
+  for (var i = 0; i < sourceFilePaths.length; i++) {
+    var folderId = this.getFolderId(sourceFilePaths[i]);
+    var folder = DriveApp.getFolderById(folderId);
+    var destination = targetFolder.createFolder(folder.getName());
+    copyFolderUtil(folder, destination);
+    folder.setTrashed(true);
+  }
+}
+
+/**
+ * Copy file API
+ * API copies all files to the target folder.
+ * @param {Array} sourceFilePaths List of local files to be copied
+ * @param {string} targetFolderPath Destination folder path
+ */
+function copyFile(sourceFilePaths, targetFolderPath) {
+  var targetFolderId;
+  // Get folder, if it doesn't exist then create it
+  try {
+    targetFolderId = this.getFolderId(targetFolderPath);
+  } catch (e) {
+    targetFolderId = this.createFolder(targetFolderPath);
+  }
+  var targetFolder = DriveApp.getFolderById(targetFolderId);
+  // Copy all files
+  for (var i = 0; i < sourceFilePaths.length; i++) {
+    var fileId = this.getFileId(sourceFilePaths[i]);
+    var file = DriveApp.getFileById(fileId);
+    file.makeCopy(file.getName(), targetFolder);
+  }
+}
+
+/**
+ * Copy folder API
+ * API copies all folders to the target folder. Apps Script does not have a
+ * direct copy folder API, folder copy is done recursively
+ * @param {Array} sourceFilePaths List of local folders to be copied
+ * @param {string} targetFolderPath Destination folder path
+ */
+function copyFolder(sourceFilePaths, targetFolderPath) {
+  var targetFolderId;
```

Comment:
The initial part of getting targetFolder is common to this and the function above so can make a helper function

Reply:
Refactored to `getOrCreateFolderUtil`